google-datacatalog-kafka-connector/src/google/datacatalog_connectors/kafka/scrape/metadata_scraper.py

```diff
@@ -0,0 +1,33 @@
+import logging
+from confluent_kafka import Consumer
+
+
+class MetadataScraper:
+
+    def __init__(self):
+        pass
+
+    def get_metadata(self, connection_args):
+        metadata = self._get_metadata_from_message_broker_connection(
+            connection_args)
+        metadata = self._create_metadata_dict(metadata)
+        return metadata
+
+    def _get_metadata_from_message_broker_connection(self, connection_args):
+        consumer = None
+        try:
+            consumer = Consumer(connection_args)
+            cluster_metadata = consumer.list_topics()
```

Comment:
list_topics_response perhaps? This doesn't contain all the cluster metadata, only the part returned by list_topics API call.

Reply:
it does return an object of class ClusterMetadata though, that's why I have called it that way. But I see what you mean - it is not transparent why it is called cluster_metadata and it can be misleading